The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5514967630/jobs/10054808096) [(plain)](https://github.com/rust-lang/rust/commit/a88a4fb6e946c7748ad9ad313770a92a678f6027/checks/14930850760/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
  lfs: false
  submodules: false
  set-safe-directory: true
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/should-skip-this.sh
src/ci/scripts/should-skip-this.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-channel.sh
src/ci/scripts/verify-channel.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/collect-cpu-stats.sh
src/ci/scripts/collect-cpu-stats.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-sccache.sh
src/ci/scripts/install-sccache.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/select-xcode.sh
src/ci/scripts/select-xcode.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-clang.sh
src/ci/scripts/install-clang.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-wix.sh
src/ci/scripts/install-wix.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-msys2.sh
src/ci/scripts/install-msys2.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-mingw.sh
src/ci/scripts/install-mingw.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-ninja.sh
src/ci/scripts/install-ninja.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/enable-docker-ipv6.sh
src/ci/scripts/enable-docker-ipv6.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/verify-line-endings.sh
src/ci/scripts/verify-line-endings.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/verify-backported-commits.sh
src/ci/scripts/verify-backported-commits.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-stable-version-number.sh
src/ci/scripts/verify-stable-version-number.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/run-build-from-ci.sh
src/ci/scripts/run-build-from-ci.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: a88a4fb6e946c7748ad9ad313770a92a678f6027
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---

error: make failed
status: exit status: 2
Build completed unsuccessfully in 0:26:02
command: cd "/checkout/tests/run-make/short-ice" && AR="ar" CC="cc -ffunction-sections -fdata-sections -fPIC -m64" CXX="c++ -ffunction-sections -fdata-sections -fPIC -m64" HOST_RPATH_DIR="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" LD_LIB_PATH_ENVVAR="LD_LIBRARY_PATH" LLVM_BIN_DIR="/usr/lib/llvm-14/bin" LLVM_COMPONENTS="aarch64 aarch64asmparser aarch64codegen aarch64desc aarch64disassembler aarch64info aarch64utils aggressiveinstcombine all all-targets amdgpu amdgpuasmparser amdgpucodegen amdgpudesc amdgpudisassembler amdgpuinfo amdgputargetmca amdgpuutils analysis arm armasmparser armcodegen armdesc armdisassembler arminfo armutils asmparser asmprinter avr avrasmparser avrcodegen avrdesc avrdisassembler avrinfo binaryformat bitreader bitstreamreader bitwriter bpf bpfasmparser bpfcodegen bpfdesc bpfdisassembler bpfinfo cfguard codegen core coroutines coverage debuginfocodeview debuginfodwarf debuginfogsym debuginfomsf debuginfopdb demangle dlltooldriver dwarflinker dwp engine executionengine extensions filecheck frontendopenacc frontendopenmp fuzzmutate globalisel hexagon hexagonasmparser hexagoncodegen hexagondesc hexagondisassembler hexagoninfo instcombine instrumentation interfacestub interpreter ipo irreader jitlink lanai lanaiasmparser lanaicodegen lanaidesc lanaidisassembler lanaiinfo libdriver lineeditor linker lto m68k m68kasmparser m68kcodegen m68kdesc m68kdisassembler m68kinfo mc mca mcdisassembler mcjit mcparser mips mipsasmparser mipscodegen mipsdesc mipsdisassembler mipsinfo mirparser msp430 msp430asmparser msp430codegen msp430desc msp430disassembler msp430info native nativecodegen nvptx nvptxcodegen nvptxdesc nvptxinfo objcarcopts object objectyaml option orcjit orcshared orctargetprocess passes perfjitevents powerpc powerpcasmparser powerpccodegen powerpcdesc powerpcdisassembler powerpcinfo profiledata remarks riscv riscvasmparser riscvcodegen riscvdesc riscvdisassembler riscvinfo runtimedyld scalaropts selectiondag sparc sparcasmparser sparccodegen sparcdesc sparcdisassembler sparcinfo support symbolize systemz systemzasmparser systemzcodegen systemzdesc systemzdisassembler systemzinfo tablegen target textapi transformutils ve veasmparser vecodegen vectorize vedesc vedisassembler veinfo webassembly webassemblyasmparser webassemblycodegen webassemblydesc webassemblydisassembler webassemblyinfo webassemblyutils windowsmanifest x86 x86asmparser x86codegen x86desc x86disassembler x86info x86targetmca xcore xcorecodegen xcoredesc xcoredisassembler xcoreinfo xray" LLVM_FILECHECK="/usr/lib/llvm-14/bin/FileCheck" NODE="/usr/bin/node" PYTHON="/usr/bin/python3" RUSTC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" RUSTDOC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" RUST_BUILD_STAGE="stage2-x86_64-unknown-linux-gnu" RUST_DEMANGLER="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools-bin/rust-demangler" S="/checkout" TARGET="x86_64-unknown-linux-gnu" TARGET_RPATH_DIR="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" TMPDIR="/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/short-ice/short-ice" "make"
bash check.sh
error: internal compiler error: no errors reported for args
##[error] --> src/lib.rs:6:5
  |
  |
6 |     func(1);
  |     ^^^^^^^

thread 'rustc' panicked at 'aborting due to `-Z treat-err-as-bug=1`', compiler/rustc_errors/src/lib.rs:1712:30
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   1: core::panicking::panic_fmt
   2: <rustc_errors::HandlerInner>::panic_if_treat_err_as_bug
   4: rustc_interface::callbacks::track_diagnostic
   5: <rustc_errors::HandlerInner>::emit_diagnostic
   6: <rustc_errors::HandlerInner>::emit_diag_at_span::<rustc_span::span_encoding::Span>
   7: <rustc_errors::HandlerInner>::span_bug::<rustc_span::span_encoding::Span, &str>
   7: <rustc_errors::HandlerInner>::span_bug::<rustc_span::span_encoding::Span, &str>
   8: <rustc_errors::Handler>::delay_span_bug::<rustc_span::span_encoding::Span, &str>
   9: <rustc_hir_typeck::fn_ctxt::FnCtxt>::report_arg_errors
  10: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_argument_types
  11: <rustc_hir_typeck::fn_ctxt::FnCtxt>::confirm_builtin_call
  12: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_call
  13: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_kind
  14: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  15: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation
  16: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_stmt
  17: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_block_with_expected
  18: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_kind
  19: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  20: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation
  21: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_return_expr
  22: rustc_hir_typeck::check::check_fn
  23: rustc_hir_typeck::typeck
  24: <core::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures::sync::par_for_each_in<&[rustc_span::def_id::LocalDefId], <rustc_middle::hir::map::Map>::par_body_owners<rustc_hir_analysis::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}> as core::ops::function::FnOnce<()>>::call_once
  25: std::panicking::try::<(), core::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures::sync::par_for_each_in<&[rustc_span::def_id::LocalDefId], <rustc_middle::hir::map::Map>::par_body_owners<rustc_hir_analysis::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}>>
  26: rustc_data_structures::sync::par_for_each_in::<&[rustc_span::def_id::LocalDefId], <rustc_middle::hir::map::Map>::par_body_owners<rustc_hir_analysis::check_crate::{closure#7}>::{closure#0}>
  28: rustc_interface::passes::analysis
  29: <rustc_interface::queries::QueryResult<&rustc_middle::ty::context::GlobalCtxt>>::enter::<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#1}::{closure#2}::{closure#4}>
  30: <rustc_interface::interface::Compiler>::enter::<rustc_driver_impl::run_compiler::{closure#1}::{closure#2}, core::result::Result<core::option::Option<rustc_interface::queries::Linker>, rustc_span::ErrorGuaranteed>>
  31: rustc_span::set_source_map::<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_interface::interface::run_compiler<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
---

note: compiler flags: -Z treat-err-as-bug=1

query stack during panic:
#0 [typeck] type-checking `main`
end of query stack
=====================
error: internal compiler error: no errors reported for args
##[error] --> src/lib.rs:6:5
##[error] --> src/lib.rs:6:5
  |
6 |     func(1);
  |     ^^^^^^^

thread 'rustc' panicked at 'aborting due to `-Z treat-err-as-bug=1`', compiler/rustc_errors/src/lib.rs:1712:30
   0:     0x7f59eb619451 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf179d8c9f784da2c
   1:     0x7f59eb6875f8 - core::fmt::write::h0c6152161bc71ce8
   2:     0x7f59eb60d501 - std::io::Write::write_fmt::h7ff22fdde828db2c
   3:     0x7f59eb619261 - std::sys_common::backtrace::print::hcb47445942aec78d
---
   8:     0x7f59eb61c899 - std::panicking::begin_panic_handler::{{closure}}::h3c731be99503c9a2
   9:     0x7f59eb619916 - std::sys_common::backtrace::__rust_end_short_backtrace::hb5a44170dab8c4e6
  10:     0x7f59eb61c5a7 - rust_begin_unwind
  11:     0x7f59eb5cd0e3 - core::panicking::panic_fmt::hbf0a646e8fa54f54
  12:     0x7f59ef2cbad7 - <rustc_errors[3a93334ba0f93ab0]::HandlerInner>::panic_if_treat_err_as_bug
  13:     0x7f59ef2caf5c - <rustc_errors[3a93334ba0f93ab0]::HandlerInner>::emit_diagnostic::{closure#2}
  14:     0x7f59ec4fa53e - rustc_interface[9ced753ac5202c90]::callbacks::track_diagnostic
  15:     0x7f59ef2ca6e2 - <rustc_errors[3a93334ba0f93ab0]::HandlerInner>::emit_diagnostic
  16:     0x7f59ecb2ec91 - <rustc_errors[3a93334ba0f93ab0]::HandlerInner>::emit_diag_at_span::<rustc_span[b8c7925c7c197b77]::span_encoding::Span>
  17:     0x7f59ecb2ee6c - <rustc_errors[3a93334ba0f93ab0]::HandlerInner>::span_bug::<rustc_span[b8c7925c7c197b77]::span_encoding::Span, &str>
  18:     0x7f59ecb2dffa - <rustc_errors[3a93334ba0f93ab0]::Handler>::delay_span_bug::<rustc_span[b8c7925c7c197b77]::span_encoding::Span, &str>
  19:     0x7f59ec8fb2d9 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::report_arg_errors
  20:     0x7f59ec8fa8b4 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_argument_types
  21:     0x7f59ec8bfbe1 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::confirm_builtin_call
  22:     0x7f59ec8be690 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_call
  23:     0x7f59ec94dd24 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_expr_kind
  24:     0x7f59ec8d9c23 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  25:     0x7f59ec94cead - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_expr_with_expectation
  26:     0x7f59ec9024f3 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_stmt
  27:     0x7f59ec902a55 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_block_with_expected
  28:     0x7f59ec94d971 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_expr_kind
  29:     0x7f59ec8d9c23 - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  30:     0x7f59ec94cead - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_expr_with_expectation
  31:     0x7f59ec8dbc8d - <rustc_hir_typeck[704b85c94a1adf17]::fn_ctxt::FnCtxt>::check_return_expr
  32:     0x7f59ecb47e31 - rustc_hir_typeck[704b85c94a1adf17]::check::check_fn
  33:     0x7f59eca7588f - rustc_hir_typeck[704b85c94a1adf17]::typeck
  34:     0x7f59edd1da37 - rustc_query_impl[ff51766b93e919dc]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ff51766b93e919dc]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f6e6804a4bb39e47]::query::erase::Erased<[u8; 8usize]>>
  35:     0x7f59ede6b4c4 - <rustc_query_impl[ff51766b93e919dc]::query_impl::typeck::dynamic_query::{closure#2} as core[98176381a533d97f]::ops::function::FnOnce<(rustc_middle[f6e6804a4bb39e47]::ty::context::TyCtxt, rustc_span[b8c7925c7c197b77]::def_id::LocalDefId)>>::call_once
  36:     0x7f59ee0845a8 - rustc_query_system[d6b63e018f0e907b]::query::plumbing::try_execute_query::<rustc_query_impl[ff51766b93e919dc]::DynamicConfig<rustc_query_system[d6b63e018f0e907b]::query::caches::VecCache<rustc_span[b8c7925c7c197b77]::def_id::LocalDefId, rustc_middle[f6e6804a4bb39e47]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ff51766b93e919dc]::plumbing::QueryCtxt, false>
  37:     0x7f59edfdd436 - rustc_query_impl[ff51766b93e919dc]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace
  38:     0x7f59ecc6e7e8 - <core[98176381a533d97f]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[da4e0ba2dba70d34]::sync::par_for_each_in<&[rustc_span[b8c7925c7c197b77]::def_id::LocalDefId], <rustc_middle[f6e6804a4bb39e47]::hir::map::Map>::par_body_owners<rustc_hir_analysis[6f7cedd4bd161642]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}> as core[98176381a533d97f]::ops::function::FnOnce<()>>::call_once
  39:     0x7f59ecc7ea46 - std[38f044fa4008a16b]::panicking::try::<(), core[98176381a533d97f]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[da4e0ba2dba70d34]::sync::par_for_each_in<&[rustc_span[b8c7925c7c197b77]::def_id::LocalDefId], <rustc_middle[f6e6804a4bb39e47]::hir::map::Map>::par_body_owners<rustc_hir_analysis[6f7cedd4bd161642]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}>>
  40:     0x7f59ecc5846f - rustc_data_structures[da4e0ba2dba70d34]::sync::par_for_each_in::<&[rustc_span[b8c7925c7c197b77]::def_id::LocalDefId], <rustc_middle[f6e6804a4bb39e47]::hir::map::Map>::par_body_owners<rustc_hir_analysis[6f7cedd4bd161642]::check_crate::{closure#7}>::{closure#0}>
  41:     0x7f59ecd86812 - rustc_hir_analysis[6f7cedd4bd161642]::check_crate
  43:     0x7f59edd1dff5 - rustc_query_impl[ff51766b93e919dc]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ff51766b93e919dc]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f6e6804a4bb39e47]::query::erase::Erased<[u8; 1usize]>>
  44:     0x7f59edeb6ae0 - <rustc_query_impl[ff51766b93e919dc]::query_impl::analysis::dynamic_query::{closure#2} as core[98176381a533d97f]::ops::function::FnOnce<(rustc_middle[f6e6804a4bb39e47]::ty::context::TyCtxt, ())>>::call_once
  45:     0x7f59edffef1e - rustc_query_system[d6b63e018f0e907b]::query::plumbing::try_execute_query::<rustc_query_impl[ff51766b93e919dc]::DynamicConfig<rustc_query_system[d6b63e018f0e907b]::query::caches::SingleCache<rustc_middle[f6e6804a4bb39e47]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[ff51766b93e919dc]::plumbing::QueryCtxt, false>
  46:     0x7f59edfa853d - rustc_query_impl[ff51766b93e919dc]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  46:     0x7f59edfa853d - rustc_query_impl[ff51766b93e919dc]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  47:     0x7f59ec382b6a - <rustc_interface[9ced753ac5202c90]::queries::QueryResult<&rustc_middle[f6e6804a4bb39e47]::ty::context::GlobalCtxt>>::enter::<core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>, rustc_driver_impl[56c425522b5f5d5e]::run_compiler::{closure#1}::{closure#2}::{closure#4}>
  48:     0x7f59ec389561 - <rustc_interface[9ced753ac5202c90]::interface::Compiler>::enter::<rustc_driver_impl[56c425522b5f5d5e]::run_compiler::{closure#1}::{closure#2}, core[98176381a533d97f]::result::Result<core[98176381a533d97f]::option::Option<rustc_interface[9ced753ac5202c90]::queries::Linker>, rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>>
  49:     0x7f59ec36f619 - rustc_span[b8c7925c7c197b77]::set_source_map::<core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>, rustc_interface[9ced753ac5202c90]::interface::run_compiler<core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>, rustc_driver_impl[56c425522b5f5d5e]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  50:     0x7f59ec33ff69 - <scoped_tls[8a82979c8fedb862]::ScopedKey<rustc_span[b8c7925c7c197b77]::SessionGlobals>>::set::<rustc_interface[9ced753ac5202c90]::interface::run_compiler<core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>, rustc_driver_impl[56c425522b5f5d5e]::run_compiler::{closure#1}>::{closure#0}, core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>>
  51:     0x7f59ec320906 - std[38f044fa4008a16b]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[9ced753ac5202c90]::util::run_in_thread_pool_with_globals<rustc_interface[9ced753ac5202c90]::interface::run_compiler<core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>, rustc_driver_impl[56c425522b5f5d5e]::run_compiler::{closure#1}>::{closure#0}, core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>>
  52:     0x7f59ec31e448 - std[38f044fa4008a16b]::panicking::try::<core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>, core[98176381a533d97f]::panic::unwind_safe::AssertUnwindSafe<<std[38f044fa4008a16b]::thread::Builder>::spawn_unchecked_<rustc_interface[9ced753ac5202c90]::util::run_in_thread_pool_with_globals<rustc_interface[9ced753ac5202c90]::interface::run_compiler<core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>, rustc_driver_impl[56c425522b5f5d5e]::run_compiler::{closure#1}>::{closure#0}, core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
  53:     0x7f59ec37817f - <<std[38f044fa4008a16b]::thread::Builder>::spawn_unchecked_<rustc_interface[9ced753ac5202c90]::util::run_in_thread_pool_with_globals<rustc_interface[9ced753ac5202c90]::interface::run_compiler<core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>, rustc_driver_impl[56c425522b5f5d5e]::run_compiler::{closure#1}>::{closure#0}, core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[98176381a533d97f]::result::Result<(), rustc_span[b8c7925c7c197b77]::ErrorGuaranteed>>::{closure#1} as core[98176381a533d97f]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  55:     0x7f59eb3beb43 - <unknown>
  56:     0x7f59eb450a00 - <unknown>
  57:                0x0 - <unknown>

---

note: compiler flags: -Z treat-err-as-bug=1

query stack during panic:
#0 [typeck] type-checking `main`
end of query stack
=====================
short backtrace: 55
full  backtrace: 79
```

</details>

--- a/compiler/rustc_hir_analysis/src/check_unused.rs
+++ b/compiler/rustc_hir_analysis/src/check_unused.rs
@@ -7,6 +7,10 @@ use rustc_session::lint;
 pub fn check_crate(tcx: TyCtxt<'_>) {
     let mut used_trait_imports: UnordSet<LocalDefId> = Default::default();
 
+    // FIXME: Use `tcx.hir().par_body_owners()` when we implement creating `DefId`s
+    // for anon constants during their parents' typeck.
+    // Doing so at current will produce queries cycle errors because it may typeck
+    // on anon constants directly.
     for item_def_id in tcx.hir().body_owners() {
         let imports = tcx.used_trait_imports(item_def_id);
         debug!("GatherVisitor: item_def_id={:?} with imports {:#?}", item_def_id, imports);

--- a/compiler/rustc_hir_analysis/src/lib.rs
+++ b/compiler/rustc_hir_analysis/src/lib.rs
@@ -116,6 +116,7 @@ use std::ops::Not;
 
 use astconv::{AstConv, OnlySelfBounds};
 use bounds::Bounds;
+use rustc_hir::def::DefKind;
 
 fluent_messages! { "../messages.ftl" }
 
@@ -498,6 +499,17 @@ pub fn check_crate(tcx: TyCtxt<'_>) -> Result<(), ErrorGuaranteed> {
     // NOTE: This is copy/pasted in librustdoc/core.rs and should be kept in sync.
     tcx.sess.time("item_types_checking", || {
         tcx.hir().for_each_module(|module| tcx.ensure().check_mod_item_types(module))
+    });
+
+    // FIXME: Remove this when we implement creating `DefId`s
+    // for anon constants during their parents' typeck.
+    // Typeck all body owners in parallel will produce queries
+    // cycle errors because it may typeck on anon constants directly.
+    tcx.hir().par_body_owners(|item_def_id| {
+        let def_kind = tcx.def_kind(item_def_id);
+        if !matches!(def_kind, DefKind::AnonConst) {
+            tcx.ensure().typeck(item_def_id);
+        }
     });
 
     check_unused::check_crate(tcx);

--- a/tests/run-make/short-ice/check.sh
+++ b/tests/run-make/short-ice/check.sh
@@ -28,8 +28,8 @@ echo "rustc_query_count_full: $rustc_query_count_full"
 ## and marks are in pairs.
 if [ $short -lt $full ] &&
     [ $begin_count -eq $end_count ] &&
-    [ $(($rustc_query_count + 10)) -lt $rustc_query_count_full ] &&
-    [ $rustc_query_count_full -gt 10 ]; then
+    [ $(($rustc_query_count + 5)) -lt $rustc_query_count_full ] &&
+    [ $rustc_query_count_full -gt 5 ]; then
     exit 0
 else
     exit 1

--- a/tests/ui/const-generics/late-bound-vars/in_closure.stderr
+++ b/tests/ui/const-generics/late-bound-vars/in_closure.stderr
@@ -6,8 +6,7 @@ error: query stack during panic:
 #4 [eval_to_allocation_raw] const-evaluating + checking `test::{closure#0}::{constant#1}`
 #5 [eval_to_valtree] evaluating type-level constant
 #6 [typeck] type-checking `test`
-#7 [used_trait_imports] finding used_trait_imports `test`
-#8 [analysis] running analysis passes on this crate
+#7 [analysis] running analysis passes on this crate
 end of query stack
 error: aborting due to previous error
 

--- a/tests/ui/const-generics/late-bound-vars/simple.stderr
+++ b/tests/ui/const-generics/late-bound-vars/simple.stderr
@@ -6,8 +6,7 @@ error: query stack during panic:
 #4 [eval_to_allocation_raw] const-evaluating + checking `test::{constant#1}`
 #5 [eval_to_valtree] evaluating type-level constant
 #6 [typeck] type-checking `test`
-#7 [used_trait_imports] finding used_trait_imports `test`
-#8 [analysis] running analysis passes on this crate
+#7 [analysis] running analysis passes on this crate
 end of query stack
 error: aborting due to previous error
 

--- a/tests/ui/const-generics/transmute-fail.stderr
+++ b/tests/ui/const-generics/transmute-fail.stderr
@@ -16,18 +16,6 @@ LL |     std::mem::transmute(v)
    = note: source type: `[[u32; H]; W]` (this type does not have a fixed size)
    = note: target type: `[[u32; W]; H]` (size can vary because of [u32; W])
 
-error[E0308]: mismatched types
-  --> $DIR/transmute-fail.rs:12:53
-   |
-LL | fn bar<const W: bool, const H: usize>(v: [[u32; H]; W]) -> [[u32; W]; H] {
-   |                                                     ^ expected `usize`, found `bool`
-
-error[E0308]: mismatched types
-  --> $DIR/transmute-fail.rs:12:67
-   |
-LL | fn bar<const W: bool, const H: usize>(v: [[u32; H]; W]) -> [[u32; W]; H] {
-   |                                                                   ^ expected `usize`, found `bool`
-
 error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> $DIR/transmute-fail.rs:23:5
    |
@@ -45,6 +33,18 @@ LL |     std::mem::transmute(v)
    |
    = note: source type: `[[[u32; 8888888]; 9999999]; 777777777]` (values of the type `[[u32; 8888888]; 9999999]` are too big for the current architecture)
    = note: target type: `[[[u32; 9999999]; 777777777]; 8888888]` (values of the type `[[u32; 9999999]; 777777777]` are too big for the current architecture)
+
+error[E0308]: mismatched types
+  --> $DIR/transmute-fail.rs:12:53
+   |
+LL | fn bar<const W: bool, const H: usize>(v: [[u32; H]; W]) -> [[u32; W]; H] {
+   |                                                     ^ expected `usize`, found `bool`
+
+error[E0308]: mismatched types
+  --> $DIR/transmute-fail.rs:12:67
+   |
+LL | fn bar<const W: bool, const H: usize>(v: [[u32; H]; W]) -> [[u32; W]; H] {
+   |                                                                   ^ expected `usize`, found `bool`
 
 error: aborting due to 6 previous errors
 

--- a/tests/ui/const-generics/type_mismatch.stderr
+++ b/tests/ui/const-generics/type_mismatch.stderr
@@ -11,18 +11,18 @@ LL | fn bar<const N: u8>() -> [u8; N] {}
    |        ^^^^^^^^^^^ required by this bound in `bar`
 
 error[E0308]: mismatched types
-  --> $DIR/type_mismatch.rs:2:11
-   |
-LL |     bar::<N>()
-   |           ^ expected `u8`, found `usize`
-
-error[E0308]: mismatched types
   --> $DIR/type_mismatch.rs:6:26
    |
 LL | fn bar<const N: u8>() -> [u8; N] {}
    |    ---                   ^^^^^^^ expected `[u8; N]`, found `()`
    |    |
    |    implicitly returns `()` as its body has no tail or `return` expression
+
+error[E0308]: mismatched types
+  --> $DIR/type_mismatch.rs:2:11
+   |
+LL |     bar::<N>()
+   |           ^ expected `u8`, found `usize`
 
 error[E0308]: mismatched types
   --> $DIR/type_mismatch.rs:6:31

--- a/tests/ui/privacy/privacy2.rs
+++ b/tests/ui/privacy/privacy2.rs
@@ -1,3 +1,5 @@
+// compile-flags: -Zdeduplicate-diagnostics=yes
+
 #![feature(start, no_core)]
 #![no_core] // makes debugging this test *a lot* easier (during resolve)
 

--- a/tests/ui/privacy/privacy2.stderr
+++ b/tests/ui/privacy/privacy2.stderr
@@ -1,22 +1,22 @@
 error[E0432]: unresolved import `bar::foo`
-  --> $DIR/privacy2.rs:17:9
+  --> $DIR/privacy2.rs:19:9
    |
 LL |     use bar::foo;
    |         ^^^^^^^^ no `foo` in `bar`
 
 error[E0603]: function import `foo` is private
-  --> $DIR/privacy2.rs:23:20
+  --> $DIR/privacy2.rs:25:20
    |
 LL |     use bar::glob::foo;
    |                    ^^^ private function import
    |
 note: the function import `foo` is defined here...
-  --> $DIR/privacy2.rs:10:13
+  --> $DIR/privacy2.rs:12:13
    |
 LL |         use foo;
    |             ^^^
 note: ...and refers to the function `foo` which is defined here
-  --> $DIR/privacy2.rs:14:1
+  --> $DIR/privacy2.rs:16:1
    |
 LL | pub fn foo() {}
    | ^^^^^^^^^^^^ consider importing it directly

--- a/tests/ui/privacy/privacy3.rs
+++ b/tests/ui/privacy/privacy3.rs
@@ -1,3 +1,5 @@
+// compile-flags: -Zdeduplicate-diagnostics=yes
+
 #![feature(start, no_core)]
 #![no_core] // makes debugging this test *a lot* easier (during resolve)
 

--- a/tests/ui/privacy/privacy3.stderr
+++ b/tests/ui/privacy/privacy3.stderr
@@ -1,5 +1,5 @@
 error[E0432]: unresolved import `bar::gpriv`
-  --> $DIR/privacy3.rs:18:9
+  --> $DIR/privacy3.rs:20:9
    |
 LL |     use bar::gpriv;
    |         ^^^^^^^^^^ no `gpriv` in `bar`

